sway/ipc-json.c

```diff
@@ -578,9 +578,14 @@ static void ipc_json_describe_view(struct sway_container *c, json_object *object
 	bool visible = view_is_visible(c->view);
 	json_object_object_add(object, "visible", json_object_new_boolean(visible));
 
+	enum sway_container_layout parent_layout = container_parent_layout(c);
+	int siblings = container_get_siblings(c)->length;
+	bool has_titlebar = c->current.border != B_PIXEL
+		|| ((parent_layout == L_STACKED || parent_layout == L_TABBED)
+			&& (!config->hide_lone_tab || siblings > 1));
```

Comment:
I think this can all be replaced with

```
bool has_titlebar = c->title_bar.tree->node.enabled;
```

Reply:
That seems to work correctly, though I ran into a regression with `hide_edge_borders --i3` (the window contents are positioned incorrectly).

Reply:
#8231